OrbitCore/CrashHandler.cpp

```diff
@@ -10,89 +10,45 @@
 #include "TcpClient.h"
 
 #ifdef _WIN32
-#include <client/windows/handler/exception_handler.h>
-
-#include <fstream>
-
-//-----------------------------------------------------------------------------
-void SendDumpInternal(const std::wstring& a_Dir, const std::wstring& a_Id);
-google_breakpad::ExceptionHandler* GHandler;
+#include <Shlwapi.h>
 
 //-----------------------------------------------------------------------------
-bool DmpFilter(void* context, EXCEPTION_POINTERS* exinfo,
-               MDRawAssertionInfo* assertion) {
-  return true;
+CrashHandler::CrashHandler(const std::wstring& dump_path) {
+  bool not_init = false;
+  if (is_init_.compare_exchange_strong(not_init, true)) {
+    Init(dump_path);
+  }
 }
 
-//-----------------------------------------------------------------------------
-bool DmpCallback(const wchar_t* dump_path, const wchar_t* minidump_id,
-                 void* context, EXCEPTION_POINTERS* exinfo,
-                 MDRawAssertionInfo* assertion, bool succeeded) {
-  PRINT_FUNC;
+void CrashHandler::Init(const std::wstring& dump_path) {
```

Comment:
Can you put everything from this Init function into the constructor?

I don't see a reason why not having this in the constructor. This would also avoid having an uninitialized object state
and helps preventing bugs.